x/supplier/types/query_validation.go

```diff
@@ -47,7 +47,43 @@ func (query *QueryAllClaimsRequest) ValidateBasic() error {
 		if filter.SessionEndHeight < 0 {
 			return ErrSupplierInvalidSessionEndHeight.Wrapf("invalid session end height for claims being retrieved %d", filter.SessionEndHeight)
 		}
+	}
+	return nil
+}
+
+func (query *QueryGetProofRequest) ValidateBasic() error {
+	// Validate the supplier address
+	if _, err := sdk.AccAddressFromBech32(query.SupplierAddress); err != nil {
+		return ErrSupplierInvalidAddress.Wrapf("invalid supplier address for proof being retrieved %s; (%v)", query.SupplierAddress, err)
+	}
+
+	// TODO_TECHDEBT: Validate the session ID once we have a deterministic way to generate it
+	if query.SessionId == "" {
+		return ErrSupplierInvalidSessionId.Wrapf("invalid session ID for proof being retrieved %s", query.SessionId)
+	}
+	return nil
+}
 
+func (query *QueryAllProofsRequest) ValidateBasic() error {
+	// TODO_TECHDEBT: update function signature to receive a context.
```

Comment:
Do we have any `ValidateBasic` functions doing this? I'd argue this isn't a TODO.

Reply:
I'm realizing that this is actually not possible as it would require modification to cosmos-sdk framework method signatures (which is not feasible IMHO). I'm just going to use `context.Background()` and remove the comment for now.